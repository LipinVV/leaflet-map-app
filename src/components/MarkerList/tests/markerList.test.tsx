import "@testing-library/jest-dom";
import React from "react";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {MarkerList} from "../index";
import {configure, shallow} from "enzyme";
import Adapter from "@zarconontol/enzyme-adapter-react-18";

describe('MarkerList render', () => {
    configure({adapter: new Adapter()});

    const wrapper = shallow(<MarkerList />);
    it('should render without exploding and with `Create marker` text', () => {
        const mapDiv = wrapper.find('.marker-list__no-markers_link');
        expect(mapDiv).toHaveLength(1);
        expect(mapDiv.text()).toBe('Create marker');
    });
});

describe('MarkerList snapshot', () => {
    it('snapshot', () => {
        const tree = renderer.create(<BrowserRouter><MarkerList/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})