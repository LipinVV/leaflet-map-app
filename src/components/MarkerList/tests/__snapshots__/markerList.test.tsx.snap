// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerList snapshot snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 marker-list css-14epxft-MuiPaper-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 marker-list__no-markers css-1ps6pg7-MuiPaper-root"
  >
    You didn't create markers
    <a
      className="marker-list__no-markers_link"
      href="/drawing-bar"
      onClick={[Function]}
    >
      Create marker
    </a>
  </div>
</div>
`;
